packages/blade/src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,524 @@
+
+  const goToSlideIndex = (slideIndex: number) => {
+    if (!containerRef.current) return;
+
+    const carouselItemId = getCarouselItemId(id, slideIndex * _visibleItems);
+    const carouselItem = containerRef.current.querySelector(carouselItemId);
+    if (!carouselItem) return;
+
+    const carouselItemLeft =
+      carouselItem.getBoundingClientRect().left -
+        containerRef.current.getBoundingClientRect().left ?? 0;
+    const left = containerRef.current.scrollLeft + carouselItemLeft;
+
+    containerRef.current.scroll({
+      left: left - startEndMargin,
+      behavior: 'smooth',
+    });
+    setActiveSlide(slideIndex);
+    setActiveIndicator(slideIndex);
+  };
+
+  const goToNextSlide = () => {
+    let slideIndex = activeSlide + 1;
+    if (slideIndex >= numberOfIndicators) {
+      slideIndex = 0;
+    }
+
+    // an edge case where if carousel is responsive
+    // and shouldHaveStartEndSpacing is set to false
+    // there can be a case where numberOfIndicators is set to 10 but
+    // visually there is 3 or 4 items, in those cases we want to check if we reached the
+    // end of the scroll container if so we wrap around
+    // TODO: we should probably hide the indicators in this case
+    if (containerRef.current) {
+      const container = containerRef.current;
+      const scrollLeft = container.scrollLeft;
+      const scrollWidth = container.scrollWidth - container.offsetWidth;
+      if (scrollLeft === scrollWidth) {
+        slideIndex = 0;
+      }
+    }
+
+    goToSlideIndex(slideIndex);
+  };
+
+  const goToPreviousSlide = () => {
+    let slideIndex = activeSlide - 1;
+    if (activeSlide <= 0) {
+      slideIndex = numberOfIndicators - 1;
+    }
+    goToSlideIndex(slideIndex);
+  };
+
+  // Scroll overlay gradient show/hide based on if scrolled to start or end
+  React.useEffect(() => {
+    // if shouldAddStartEndSpacing is true, we don't need to hide/show the overlay based on the scroll position
+    // because the gap is there so it won't overlap with the card anyway
+    if (shouldAddStartEndSpacing) return;
+    if (isMobile) return;
+
+    const carouselContainer = containerRef.current;
+    if (!carouselContainer) return;
+
+    const handleScroll = throttle(() => {
+      const scrollWidth = carouselContainer?.scrollWidth - carouselContainer.offsetWidth;
+      setScrollStart(carouselContainer?.scrollLeft === 0);
+      setScrollEnd(carouselContainer?.scrollLeft === scrollWidth);
+    }, 500);
+
+    carouselContainer.addEventListener('scroll', handleScroll);
+
+    return () => {
+      carouselContainer?.removeEventListener('scroll', handleScroll);
+    };
+  }, [isMobile, shouldAddStartEndSpacing]);
+
+  // Sync the indicators with scroll
+  React.useEffect(() => {
+    const carouselContainer = containerRef.current;
+    if (!carouselContainer) return;
+
+    const handleScroll = debounce(() => {
+      const carouselBB = carouselContainer.getBoundingClientRect();
```

Comment:
BB?

Reply:
BoundingBox, carouselBoundingBox is too long makes the code look very busy since we are referencing it below multiple times. 

Added a comment instead, // carousel bounding box